src/translators/sql_translator.py

```diff
@@ -170,16 +170,46 @@ def insert(self, entities, fiware_service=None, fiware_servicepath=None):
             msg = "Entities expected to be of type list, but got {}"
             raise TypeError(msg.format(type(entities)))
 
-        entities_by_type = {}
-        for e in entities:
-            entities_by_type.setdefault(e['type'], []).append(e)
-
-        res = None
-        for et in entities_by_type.keys():
-            res = self._insert_entities_of_type(et,
-                                                entities_by_type[et],
-                                                fiware_service,
-                                                fiware_servicepath)
+        service_paths = []
+        if fiware_servicepath:
+            clean_fiware_servicepath = fiware_servicepath.replace(" ", "")
+            service_paths = clean_fiware_servicepath.split(",")
+        else:
+            service_paths = [fiware_servicepath]
+        if len(service_paths) == 1:
+            entities_by_type = {}
+            for e in entities:
+                entities_by_type.setdefault(e['type'], []).append(e)
+
+            res = None
+            for et in entities_by_type.keys():
+                res = self._insert_entities_of_type(et,
+                                                    entities_by_type[et],
+                                                    fiware_service,
+                                                    service_paths[0])
+        elif len(service_paths) == len(entities):
+            entities_by_service_path = {}
+            for idx, path in enumerate(service_paths):
+                entities_by_service_path.setdefault(path, []).append(
+                    entities[idx])
+            res = None
+            for path in entities_by_service_path.keys():
+                entities_by_type = {}
+                for e in entities_by_service_path[path]:
+                    entities_by_type.setdefault(e['type'], []).append(e)
+
+                res = None
+                for et in entities_by_type.keys():
+                    res = self._insert_entities_of_type(et,
+                                                        entities_by_type[et],
+                                                        fiware_service,
+                                                        path)
+        else:
+            msg = 'Multiple servicePath are allowed only ' \
+                  'if their size is match the size of entities'
+            raise InvalidHeaderValue('Fiware-ServicePath',
+                                     fiware_servicepath, msg)
+
```

Comment:
what happens if there's only one entity and an empty service path?

    service_paths = ['']
    for idx, path in enumerate(service_paths):
        print(f"{idx}: {path}")

    >>>     0:

so I think we'd be inserting the entity with an empty path? In general, do we still allow clients not to specify a service path, so e.g. a notification w/ 3 entities and no service path results in all three entities still being inserted under the empty path?

Reply:
may the correct one is `service_paths = [None]` 
all actually boils down to what orion provide us, right?

Reply:
see line 178, so if no service path, we exactly create an array with `service_paths = [None]`

Reply:
yep, line 178 is what makes a list with one entry if there's no service path (`None` or `''`). just wanted to double check this was intentional.

> boils down to what orion provide us, right?

yep, you're right. I don't think we've got much of a choice here. If the notification contains no service path probably we should still insert the entity. Looking at the code I think we do that in the case of one entity, but what does Orion do when notifying multiple entities with no service path? Will the notification contain a list of empty service paths, e.g. for 3 entities

    path = ' , , '
    entities = [e1, e2, e3]

or no path at all, e.g.

    path = ''
    entities = [e1, e2, e3]

If I read the code correctly, in the first case we insert all 3 entities under the empty path whereas in the second case we raise an `InvalidHeaderValue` exception. If this is the case, then my question is: do we have to insert entities if there's no service path or it is empty?

Reply:
Multiple "" will work if they are in the same number as entities.